### ISSUE TITLE
about clip score, clip temporal and FVD reprouction?

### ISSUE THREAD

**Issue by Typhoona** [AlonzoLeeeooo/StableV2V#24]
Hello, I have reproduced the CLIP score and CLIP temporal results, and they seem quite strange. Additionally, the FVD (Fréchet Video Distance) results are also not as expected. I used the code directly from the GitHub repository for the calculations. However, I noticed that the FVD results I obtained are not as good as the baseline results you provided. Could you possibly share the code for calculating these metrics? Or could you provide some insights into what might be causing these discrepancies? Or could you give me some instructions? Thank you very much, and I look forward to your reply.

(1)here's my clip temporal code:
model = CLIPModel.from_pretrained("./clip-vit-base-patch32/models--openai--clip-vit-base-patch32").to(device)
processor = CLIPProcessor.from_pretrained("./clip-vit-base-patch32/models--openai--clip-vit-base-patch32")

def calculate_clip_text(fames_root, prompt):
    """
    Calculate CLIP-Text score (text alignment)

    Args:
        video_frames: List of PIL Images (video frames)
        prompt: Text prompt used for video editing

    Returns:
        Average cosine similarity between text embeddings and frame embeddings
    """
    # Convert PIL Images to pixel values
    # print(len(video_frames),video_frames[0])#16
    images = []
    for filename in os.listdir(fames_root):
      
        if filename.endswith('.png'):
           
            file_path = os.path.join(fames_root, filename)
           

            with Image.open(file_path) as img:
                img = img.convert('RGB')
                images.append(img)

    # images = [frame.convert('RGB') for frame in video_frames]
    inputs = processor(text=[prompt], images=images, return_tensors="pt", padding=True)
    inputs = {k: v.to(device) for k, v in inputs.items()}

    # Get embeddings
    with torch.no_grad():
        outputs = model(**inputs)

    # Calculate cosine similarities
    text_emb = outputs.text_embeds
    image_embs = outputs.image_embeds
    similarities = F.cosine_similarity(text_emb, image_embs)

    return similarities.mean().item()

(2)here's my clip score code:
def calculate_clip_image(fames_root):
    """
    Calculate CLIP-Image score (temporal consistency)

    Args:
        video_frames: List of PIL Images (video frames)

    Returns:
        Average cosine similarity between consecutive frame embeddings
    """
    if len(fames_root) < 2:
        raise ValueError("At least two frames are required to calculate temporal consistency.")
    images = []
    for filename in os.listdir(fames_root):       
        if filename.endswith('.png'):            
            file_path = os.path.join(fames_root, filename)          

            with Image.open(file_path) as img:
                img = img.convert('RGB')
                images.append(img)
                
    
    # print(len(images))#16
    # Convert PIL Images to pixel values
    # images = [frame.convert('RGB') for frame in video_frames]
    inputs = processor(images=images, return_tensors="pt", padding=True)
    inputs = {k: v.to(device) for k, v in inputs.items()}

    # Get embeddings
    with torch.no_grad():
        image_features = model.get_image_features(pixel_values=inputs["pixel_values"])

    # Calculate pairwise similarities between consecutive frames
    similarities = []
    for i in range(len(images) - 1):
        emb1 = image_features[i].unsqueeze(0)
        emb2 = image_features[i + 1].unsqueeze(0)
        sim = F.cosine_similarity(emb1, emb2)
        similarities.append(sim.item())

    return np.mean(similarities)

**Comment by AlonzoLeeeooo** [comment]
Hi @Typhoona ,

Thanks for your attention! I might need some time to clean up the code for different metrics. If possible, I'll do this in near future and release the scripts in this repository.

As for the potential discrepancy during reproduction, it might be possible for the performance to fluctuate if the first frames are in bad quality. But if you use the suggested image editors in our paper, the first frames will be fine in most cases.

From you given code scripts, I've noticed that you are using `clip-vit-base-patch32`. Please note that we use [`clip-vit-large-patch14`](https://huggingface.co/openai/clip-vit-large-patch14). I think this might be an issue to cause the discrepancy.

I'll provide some of the source codes that we modify from for you if you are in an urgent need to evaluate your experiments:
- FVD: [https://github.com/JunyaoHu/common_metrics_on_video_quality](https://github.com/JunyaoHu/common_metrics_on_video_quality)
- DOVER: [https://github.com/VQAssessment/DOVER](https://github.com/VQAssessment/DOVER)
- CLIP-Temporal, CLIP scores: [https://github.com/Taited/clip-score/blob/master/src/clip_score/clip_score.py](https://github.com/Taited/clip-score/blob/master/src/clip_score/clip_score.py)
- Warping Error: [https://github.com/phoenix104104/fast_blind_video_consistency](https://github.com/phoenix104104/fast_blind_video_consistency)

Hope the above information address your concerns. If you have any other problems about reproduction, please do not hesitate to contact us again in this issue.

Best regards

**Comment by Typhoona** [comment]
Thanks a lot, I'll retry doing evaluation.

**Comment by AlonzoLeeeooo** [comment]
> Thanks a lot, I'll retry doing evaluation.

Great!
